Sim_To_DuT_Interface/DuTLogger/DuTLogger.cpp

```diff
@@ -0,0 +1,339 @@
+        quill::start();
+
+        // remember that we started the engine and checked everything
+        startedQuillEngine = true;
+    }
+}
+
+/**
+ * Creates a handler to write messages in the console.
+ * This handler can be used to connect it to a logger.
+ *
+ * @return a hadler to log messages in the console
+ */
+quill::Handler* DuTLogger::buildConsoleHandler() {
+    // build a handler for the console
+    quill::Handler* newHandler = consoleHandler = quill::stdout_handler("consoleHandler");
+    newHandler->set_log_level(DEFAULT_CONSOLE_LOG_LEVEL);
+
+    // modify the pattern for the logger
+    newHandler->set_pattern(QUILL_STRING("%(ascii_time)  %(level_name): %(message)"),
+                                "%D %H:%M:%S.%Qms");
+    return newHandler;
+}
+
+/**
+ * Creates a handler to write messages in a file.
+ * This handler can be used to connect it to a logger.
+ * It will create a file for this day, if it hasn't been done yet.
+ *
+ * @return a hadler to log messages in a file
+ */
+quill::Handler* DuTLogger::buildFileHandler() {
+    // a second handler for the file is needed
+    std::string basicPath = currentLogpathConsole + "/Logfile.log";
+    quill::Handler* newHandler = quill::file_handler(basicPath, FILE_MODE_CONSOLE,
+                                                     quill::FilenameAppend::Date);
+    newHandler->set_log_level(DEFAULT_FILE_LOG_LEVEL);
+
+    // modify the pattern for the logger
+    newHandler->set_pattern(QUILL_STRING("%(ascii_time)  %(level_name): %(message)"),
+                            "%D %H:%M:%S.%Qms");
+    // return the new handler
+    return newHandler;
+}
+
+/**
+ * Builds a logger for the console. If necessary the logger will be build with an additional handler for
+ * files. This will be configured by the second argument.
+ *
+ * @param name the name for the logger
+ * @param withFileHandler true if the logger also needs a handler for logfiles
+ * @return the created logger
+ */
+quill::Logger* DuTLogger::createConsoleLogger(const char* name, bool withFileHandler) {
+    // check if the quill engine is started
+    startEngine();
+
+    // check if the new logger needs to write the console log to a file
+    // if it needs it, we have to build the logger a different way
+    quill::Logger* newLogger;
+    if (withFileHandler) {
+        newLogger = quill::create_logger(name, {consoleHandler, consoleFileHandler});
+
+        // check if we might have to delete old logging folders
+        removeOldLogfiles(currentLogpathConsole);
+    } else {
+        newLogger = quill::create_logger(name, consoleHandler);
+    }
+
+    // Set the LogLevel (L3 for everything)
+    newLogger->set_log_level(quill::LogLevel::TraceL3);
+
+    return newLogger;
+}
+
+/**
+ * Creates a logger to log a specific data object (like events; not normal messages!) into a file.
+ * The logger will be build with all necessary handlers.
+ * It will create a file for this day, if it hasn't been done yet.
```

Comment:
We had discussed every new start of the simulation creates a new file

Reply:
I understood that we collect all logs in one file for one day. You can watch the timestamp of each message in the file to notice the start of the new simulation. But it's ok. In the header of the logger is an attribute to switch between the modes "start a new file on every simulation" and "try to append if a file for this day is there". So I will set the first mode to default setting.